src/app/model/ChessBoard.ts

```diff
@@ -0,0 +1,20 @@
+import { PiecePositions } from './PiecesPositions';
+import { Piece } from './Piece';
+import { SquareWithPiece, Square } from './Types';
+import { PieceMoves } from './PieceMoves';
+
+export class ChessBoard implements PieceMoves, PiecePositions {
+  constructor(private boardWithPieces: SquareWithPiece) {}
+
+  onPositionPiece(square: Square): Piece | undefined {
+    // if (!this.boardWithPieces[`${square.column}${square.row}`]) {
+    //   return this.boardWithPieces[`${square.column}${square.row}`]
+    // }
+    // return;
+    return this.boardWithPieces[`${square.column}${square.row}`] ? this.boardWithPieces[`${square.column}${square.row}`] : undefined;
```

Comment:
@nowakprojects jak po tym iterować ^_^ ?

Reply:
@Szambelan To już było zaimplementowane w tescie ;P 
No czekaj, po co chcesz iterować. Musisz zrozumieć co to jest boardWithPieces.
Obiekt w stylu: 
boardWithPieces = {
A1: new Rook(WHITE)
A2: new Rook(BLACK)
}

To jak mozesz sie dostac do pola obiektu boardWithPieces, które nazywa się A1 (w przypadku jak chcesz dostac z Square, ktory oznacza A1)? 
Jesli nie bedzie pola o nazwie A1 to co dostajesz?

Reply:
chyba dalej nie kumam, że to działa co napisałem ^_^ Trochę dziwne ...